pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -245,7 +268,7 @@ func (r *hibernationReconciler) Reconcile(ctx context.Context, request reconcile
 	}
 
 	if !shouldHibernate {
-		if hibernatingCondition == nil || hibernatingCondition.Status == corev1.ConditionFalse {
+		if hibernatingCondition.Status == corev1.ConditionFalse {
```

Comment:
Might be the source of some of your bugs here, the nil check should be replaced with an Unknown check.